app/Repositories/Doctrine/ORM/AccountRepository.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Repositories\Doctrine\ORM;
+
+use App\Database\Entities\Account;
+use App\Exceptions\InvalidSubscriptionTypeException;
+use App\Repositories\Interfaces\AccountRepositoryInterface;
+
+final class AccountRepository extends AbstractRepository implements AccountRepositoryInterface
+{
+    /**
+     * Allowed subscription types
+     *
+     * @var mixed[]
+     */
+    public static $subscriptionTypes = ['monthly', 'lifetime'];
+    
+    /**
+     * Find by subscription type
+     *
+     * @param string $subscriptionType
+     *
+     * @return mixed[]
+     *
+     * @throws \Exception
+     */
+    public function findBySubscriptionType(string $subscriptionType): array
+    {
+        if (\in_array(\strtolower($subscriptionType), self::$subscriptionTypes) === false) {
```

Comment:
Based on our coding standards we must ensure that we compare not only the value but also the type. Please make sure that the in_array php function compares data types too. By adding a third parameter in in_array function with value `true` we can now check the types.